apps/web/components/AdditionalCalendarSelector.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import Select from "react-select";
+import { OptionProps } from "react-select";
+
+import { InstallAppButton } from "@calcom/app-store/components";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { App } from "@calcom/types/App";
+import { Button } from "@calcom/ui";
+
+import { QueryCell } from "@lib/QueryCell";
+import { trpc, inferQueryOutput } from "@lib/trpc";
+
+interface Props {
+  isLoading?: boolean;
+}
+
+const ImageOption = (props: OptionProps) => {
+  const data = props.data as { [key: string]: string; type: App["type"] };
+  return (
+    <InstallAppButton
+      type={data.type}
+      render={(props) => (
+        <Button {...props} color="minimal" className="w-full">
+          {/* eslint-disable @next/next/no-img-element */}
+          {data.image && <img className="float-left mr-3 inline h-5 w-5" src={data.image} alt={data.label} />}
+          <p>{data.label}</p>
+        </Button>
+      )}
+    />
+  );
+};
+
+type AppOutput = inferQueryOutput<"viewer.integrations">["items"][0];
+
+const AdditionalCalendarSelector = ({ isLoading }: Props): JSX.Element | null => {
```

Comment:
New selector for additional connections

Reply:
Seems like the Apple Calendar Install Button returns the dialog to be opened, which never gets a chance to be opened as the dropdown list where the install button is included (as well as the dialog component) gets closed upon option selection. Tried a bunch of ways, got very very close, but I think the code would be too hard to follow even if we can make it work. Any suggestions on how to handle the extra element returned by the Install Button? Would it be too bad of a UX redirecting the user to some place with the dialog opened as a page? Or perhaps we can make it so the dialog becomes some kind of tooltip in this dropdown case 🤔 @Jaibles your input would be highly appreciated!

Reply:
For now it would be better to redirect to `/apps/apple-calendar` than doing nothing at all...

Reply:
Wouldn't work, the app is already installed, so the button is disabled to be able to click it to add/connect a new Apple Calendar:
<img width="1081" alt="Cal_com___Cal_com" src="https://user-images.githubusercontent.com/467258/169672477-7d1aac91-2121-451a-b075-0ee3749488a4.png">

Reply:
We need to fix the dialog to merge then